### PR TITLE
Readded intersperse and foldr test

### PR DESCRIPTION
**Pull request by radrow**
After https://github.com/aeternity/aeternity/pull/2714 and https://github.com/aeternity/aeternity/pull/2650 these test cases should be working

#### edit

Waits for https://github.com/aeternity/aesophia/pull/136 to be merged and included

### CODE REVIEW

**Comment by tolbrino** [comment]
ping @radrow The blocking PR is merged.

**Comment by radrow** [comment]
I have a break, will take care of it in January

**Comment by tolbrino** [comment]
> I have a break, will take care of it in January

One month later :-) Can you finish this one?

**Comment by radrow** [comment]
Isn't it finished? It changes only one contract and the failing CIs are failing because errors like 404 or swagger which was not touched

**Comment by tolbrino** [comment]
I guess you have to rebase on the current `master`. As long as the tests fail its not mergable.

**Comment by tolbrino** [comment]
Tests are still failing.

**Comment by radrow** [comment]
The only failures I can see are 
```
%%% aeoracle_SUITE ==> {tc_auto_skip,
    {failed,
        {aeoracle_SUITE,init_per_testcase,
            {{already_started,<0.16238.0>},
             [{meck_proc,start,
                  [aec_governance,[passthrough]],
                  [{file,
                       "/home/builder/aeternity/_build/test/lib/meck/src/meck_proc.erl"},
                   {line,97}]},
              {aeoracle_SUITE,init_per_testcase,2,
                  [{file,
                       "/home/builder/aeternity/apps/aeoracle/test/aeoracle_SUITE.erl"},
                   {line,103}]},
              {test_server,do_init_per_testcase,2,
                  [{file,"test_server.erl"},{line,1524}]},
              {test_server,run_test_case_eval1,6,
                  [{file,"test_server.erl"},{line,1225}]},
              {test_server,run_test_case_eval,9,
                  [{file,"test_server.erl"},{line,1195}]}]}}}}
```
and similar. They are clearly unrelated to the PR. I have rebased just in case

**Comment by radrow** [comment]
timeout...

File: test/contracts/stdlib_tests.aes [hunk 1]
```diff
@@ -3,6 +3,7 @@
 include "Func.aes"
 include "Option.aes"
 include "List.aes"
+include "ListInternal.aes"
```
Comment (hanssv) [inline test/contracts/stdlib_tests.aes]: Not necessary?

Comment (radrow) [reply test/contracts/stdlib_tests.aes]: I am not sure actually. It was added for some reason...

Comment (hanssv) [reply test/contracts/stdlib_tests.aes]: It compiles just fine without it

**Pull request merged**